qiskit_nature/properties/grouped_property.py

```diff
@@ -49,6 +50,14 @@ def __init__(self, name: str) -> None:
         """
         super().__init__(name)
         self._properties: dict[str, T_co] = {}
+        warn_deprecated(
+            "0.5.0",
+            old_type=DeprecatedType.CLASS,
+            old_name="qiskit_nature.properties.GroupedProperty",
+            new_type=DeprecatedType.CLASS,
+            new_name="qiskit_nature.second_q.properties.GroupedProperty",
```

Comment:
This class is actually going away in its entirety in favor of storing the properties directly on the `BaseProblem` objects.
Maybe we can refer to #705 for more information?